localstack/services/apigateway/patches.py

```diff
@@ -332,6 +332,7 @@ def create_rest_api(fn, self, *args, tags=None, **kwargs):
         """
         tags = tags or {}
         result = fn(self, *args, tags=tags, **kwargs)
+        # TODO: lower the custom_id when getting it from the tags, as AWS is case insensitive
```

Comment:
AWS doesn't have this concept of `custom_id`, this is a *feature* localstack has to facilitate lookups. You probably mean tags in general are case-insensitive. 

I do have a general sentiment, we should make these lookups case-insensitive but keep the original form. So logically would never fail (because of the casing) but keep visuals as the author intended.

Reply:
https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html

check the tag keys example and the spreadsheets headers.

Reply:
Yes, the issue is, the ID will always be lowercase when part of the host. That's why I thought we should force the restApiId to be lower case, even though the user specified a cased tag. (tag could be cased, but not the ID once used to create the restApi). 
Even though it's a LocalStack feature, a regular restApiId is always lowercase, and AWS will fail if you upper case an existing ID. Making it case insensitive changes that behaviour.

We could also simply raise an error if the user provided tag value is not lowercase for `__custom_id__` or something. 
My general idea is just that `restApiId` should be lowercase. There are no changes yet in behaviour anyway, as I keep it in a `CaseInsentiveDict`, but it makes the lookup more complicated for no real gain on our side.